Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -264,19 +243,48 @@ private static LookupElement createLookupElement(@NotNull final Project project,
         shouldSetSelection = false;
         final List<String> parameterNames = suggestion.getParameterNames();
         if (parameterNames != null) {
-          lookup = lookup.withInsertHandler(new InsertHandler<LookupElement>() {
-            @Override
-            public void handleInsert(InsertionContext context, LookupElement item) {
-              if (parameterNames.isEmpty()) {
-                ParenthesesInsertHandler.NO_PARAMETERS.handleInsert(context, item);
-              }
-              else {
-                ParenthesesInsertHandler.WITH_PARAMETERS.handleInsert(context, item);
-                // Show parameters popup.
-                final Editor editor = context.getEditor();
-                final PsiElement psiElement = lookupObject.getElement();
-                AutoPopupController.getInstance(project).autoPopupParameterInfo(editor, psiElement);
+          lookup = lookup.withInsertHandler((context, item) -> {
+            if (parameterNames.isEmpty()) {
+              ParenthesesInsertHandler.NO_PARAMETERS.handleInsert(context, item);
+            }
+            else {
+              ParenthesesInsertHandler.WITH_PARAMETERS.handleInsert(context, item);
+              // Show parameters popup.
+              final Editor editor = context.getEditor();
+              final PsiElement psiElement = lookupObject.getElement();
+
+              // Insert argument defaults if provided.
+              final String argumentListString = suggestion.getDefaultArgumentListString();
+              if (argumentListString != null) {
+                final Document document = editor.getDocument();
+                int offset = context.getSelectionEndOffset() - 1;
```

Comment:
At this point caret is already between parens, you can take `editor.getCaretModel().getOffset();`.
With the current code it is possible to get malformed code when completing over the existing text using Tab:
![image](https://cloud.githubusercontent.com/assets/4499789/23563465/d6c7123c-0057-11e7-91b5-f454dca5aee4.png)